src/test/java/com/zachary_moore/runner/LintRunnerShould.java

```diff
@@ -82,11 +87,188 @@ public void lintRunnerShouldTakeMultipleFiles() throws Exception {
-        Map<LintRule, Map<JSONFile, List<String>>> lintOutput = lintRunner.lint();
+                    "./src/test/resources/test-2.json",
+                    "./src/test/resources/test-file.pdsc");
+        lintRunner.lint();
+
+        Map<LintRule, Map<JSONFile, List<Error>>> lintOutput = lintRunner.getLintOutput();
 
-        assert(lintRunner.analyzeLintAndGiveExitCode() == 1);
+        assert(lintRunner.getExitCode() == 1);
         assert(lintOutput.get(lintRule).size() == 2);
     }
+
+    @Test
+    public void lintRunnerShouldValidErrorForWrappedPrimitive() throws Exception {
+        LintRule lintRule = new LintRule.Builder()
+            .setImplementation(new LintImplementation<WrappedPrimitive<String>>() {
+                @Override
+                public Class<?> getClazz() {
+                    return String.class;
+                }
+
+                @Override
+                public boolean shouldReport(WrappedPrimitive<String> stringWrappedPrimitive) {
+                    this.setReportMessage(stringWrappedPrimitive.toString());
+                    return true;
+                }
+            })
+            .setIssueId("")
+            .setLevel(LintLevel.ERROR)
+            .build();
+
+        LintRegister lintRegister = new LintRegister();
+        lintRegister.register(lintRule);
+        LintRunner lintRunner =
+            new LintRunner(
+                lintRegister,
+                "./src/test/resources/test-3.json");
+        lintRunner.lint();
+
+        Map<LintRule, Map<JSONFile, List<Error>>> lintOutput = lintRunner.getLintOutput();
+
+        assert(lintRunner.getExitCode() == 1);
+        assert(lintOutput.get(lintRule).size() == 1);
+
+        List<String> exitingPaths = new ArrayList<>();
+        lintOutput.get(lintRule).values().forEach(k -> k.forEach(k1 -> exitingPaths.add(k1.getPath())));
+        assert(exitingPaths.size() == 4);
+
+        List<String> validPaths = Arrays.asList(
+            "test_1_label",
+            "test_2_label",
+            "test_3_label.test_3_1_label.test_3_1_1_label.test_3_1_1_1_label",
+            "test_3_label.test_3_2_label.test_3_2_1_label");
+        assert(exitingPaths.containsAll(validPaths));
+    }
+
+    @Test
+    public void lintRunnerShouldValidErrorForJSONObject() throws Exception {
+        LintRule lintRule = new LintRule.Builder()
+            .setImplementation(new LintImplementation<JSONObject>() {
+                @Override
+                public Class<?> getClazz() {
+                    return JSONObject.class;
+                }
+
+                @Override
+                public boolean shouldReport(JSONObject jsonObject) {
+                    this.setReportMessage(jsonObject.toString());
+                    return true;
+                }
+            })
+            .setIssueId("")
+            .setLevel(LintLevel.ERROR)
+            .build();
+
+        LintRegister lintRegister = new LintRegister();
```

Comment:
can we re-use these across tests vs making them in each ? Would make test code a lot cleaner